lib/charms/tempo_k8s/v1/charm_tracing.py

```diff
@@ -127,11 +127,32 @@ def get_current_span() -> Union[Span, None]:
     return cast(Span, span)
 
 
+def _get_tracer_from_context(ctx: Context) -> Optional[ContextVar]:
+    try:
+        return next(filter(lambda v: v is not None and v.name == "tracer", ctx.keys()))
```

Comment:
I know I'm guilty of using this pattern a bit too often myself, but I get reminded almost weekly that it's more readable to use a list comprehension instead.
```suggestion
    tracers = [v for v in ctx if v is not None and v.name == "tracer"]
    if tracers:
        return tracers[0]
        
```

Reply:
Good point, fixed.